tests/unit/test_utils.py

```diff
@@ -554,6 +668,89 @@ def test_callback_will_also_be_triggered_by_seek(self):
         chunk.read(2)
         self.assertEqual(self.amounts_seen, [2, -2, 2, -1, 2])
 
+    def test_callback_triggered_by_out_of_bound_seeks(self):
+        data = b'abcdefghij1234567890klmnopqr'
+
+        # Create test file
+        filename = os.path.join(self.tempdir, 'foo')
+        with open(filename, 'wb') as f:
+            f.write(data)
+        chunk = ReadFileChunk.from_filename(
+            filename, start_byte=10, chunk_size=10,
+            callbacks=[self.callback])
+
+        expected_callback_prog = [10, -5, 5, -1, 1, -1, 1, -5, 5, -10]
+
+        self._assert_out_of_bound_start_seek(chunk, expected_callback_prog)
+        self._assert_out_of_bound_relative_seek(chunk, expected_callback_prog)
+        self._assert_out_of_bound_end_seek(chunk, expected_callback_prog)
+
+    def _assert_out_of_bound_start_seek(self, chunk, expected):
+        # clear amounts_seen
+        self.amounts_seen = []
+        self.assertEqual(self.amounts_seen, [])
+
+        # (position, change)
+        chunk.seek(20) # (20, 10)
+        chunk.seek(5) # (5, -5)
+        chunk.seek(20) # (20, 5)
+        chunk.seek(9) # (9, -1)
+        chunk.seek(20) # (20, 1)
+        chunk.seek(11) # (11, 0)
+        chunk.seek(20) # (20, 0)
```

Comment:
In these comments we should note that they will not be in the `expected` list because `invoke_progress_callbacks` is called. That threw me for a bit trying to understand why `0` was not in the list.